lms/views/api/canvas/files.py

```diff
@@ -33,16 +41,19 @@ def via_url(self):
         application_instance = self.request.find_service(
             name="application_instance"
         ).get()
-
         assignment = self.request.find_service(name="assignment").get(
             application_instance.tool_consumer_instance_guid,
             self.request.matchdict["resource_link_id"],
         )
 
+        # We don't use the assignment document_url directly as it might be out of sync with what's stored in canvas
+        document_url = self.request.params["document_url"]
```

Comment:
This view always takes the document_url param.

It will up to the callers, the basic_lti_views to decide where that URL comes from, directly from the DB row or built form the query params for legacy ones.

Reply:
This does kind of leave me wondering why we don't use the `document_url` from the DB? If the initial LTI launch  would always update the `ModuleItemConfiguration` in the DB with the latest `document_url` then by the time we get to this `via_url` API call we would know that the DB is up to date.

In the case of Blackboard files the `via_url` API doesn't need to query the DB at all. It just receives the `document_url` in the query params and uses it. But for Canvas files, because of course copy, the `via_url` has to look up the assignment in the DB anyway. At which point it may as well use the `document_url` from the assignment. Then no `document_url` query param would be needed.